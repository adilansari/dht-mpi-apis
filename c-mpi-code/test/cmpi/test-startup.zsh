#!/bin/zsh

# Be sure to make tests with D=1

# set -x

source tools/test-helpers.zsh

PROGRAM=$1
NODES=$2
CLIENTS=$3
shift 3
LAUNCH=${*}

OUTPUT=${PROGRAM%.x}.out
TOTAL=$(( NODES+CLIENTS ))

if (( USE_COMM_WORLD == 1 ))
then

  eval ${LAUNCH} -n ${TOTAL} ${PROGRAM} -n ${NODES} -p reps=10 >& ${OUTPUT}
  CODE=$?
  [[ ${CODE} == 0 ]] || crash "exit code was: ${CODE}"

else

  # KDA-2B execution:

  tools/startup_nodes.zsh $(( N-1 )) 100 &

  mpiexec -n 1 test/cmpi/test01.x > CLIENT.out

  sleep $(( N+15 ))

fi

# Should be TOTAL "Normal exit."s
N=$( grep -c "Normal exit." ${OUTPUT} )
(( N == TOTAL )) || crash "N != ${TOTAL}"

exit 0
