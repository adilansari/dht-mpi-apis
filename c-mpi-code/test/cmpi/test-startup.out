Debug to stdout
START: at 1386358582.127176 
[2] cmpi_params_add(key=nodes,value=5)... 
[0] cmpi_params_add(key=nodes,value=5)... 
[5] cmpi_params_add(key=nodes,value=5)... 
[0] cmpi_params_add(key=reps,value=10)... 
[2] cmpi_params_add(key=reps,value=10)... 
[5] cmpi_params_add(key=reps,value=10)... 
[1] cmpi_params_add(key=nodes,value=5)... 
[3] cmpi_params_add(key=nodes,value=5)... 
[1] cmpi_params_add(key=reps,value=10)... 
[3] cmpi_params_add(key=reps,value=10)... 
[4] cmpi_params_add(key=nodes,value=5)... 
[4] cmpi_params_add(key=reps,value=10)... 
[0] cmpi_params_get(key=tag,result=(null))... 
[0] I am node 
node: 0 @ roxy
[2] cmpi_params_get(key=tag,result=(null))... 
[2] I am node 
node: 2 @ roxy
[3] cmpi_params_get(key=tag,result=(null))... 
[3] I am node 
node: 3 @ roxy
[4] cmpi_params_get(key=tag,result=(null))... 
[4] I am node 
node: 4 @ roxy
[D 14:36:23.139891 R:0] node: 0
[0] cmpi_params_get(key=nodes,result=5)... 
[D 14:36:23.139904 R:3] node: 3
[3] cmpi_params_get(key=nodes,result=5)... 
[3] DENSE_Init()... 
[0] DENSE_Init()... 
[0] rpc_bootping(1)... 
[0] MPIRPC_Block(bootping)... 
[0] MPIRPC_Block_blob(bootping)... 
[3] listen_loop()... 
[0] MPIRPC_Call_blob(target.rank=1,name=bootping,args=(null),rpc->unique=1)... 
[5] cmpi_params_get(key=tag,result=(null))... 
[D 14:36:23.140206 R:4] node: 4
[D 14:36:23.140085 R:2] node: 2
[2] cmpi_params_get(key=nodes,result=5)... 
[2] DENSE_Init()... 
[1] cmpi_params_get(key=tag,result=(null))... 
[1] I am node 
node: 1 @ roxy
[2] listen_loop()... 
[4] cmpi_params_get(key=nodes,result=5)... 
[4] DENSE_Init()... 
[4] listen_loop()... 
[5] I am client 
client: 5 @ roxy
[D 14:36:23.140770 R:5] client: 5
[5] cmpi_params_get(key=nodes,result=5)... 
[5] contact: 0
[5] contact: 1
[5] contact: 2
[5] contact: 3
[5] contact: 4
[5] cmpi_client_code()... 
sleeping...
[D 14:36:23.142523 R:1] node: 1
[1] cmpi_params_get(key=nodes,result=5)... 
[1] DENSE_Init()... 
[1] listen_loop()... 
[1] MPIRPC_Request(name=bootping,caller.rank=0)... 
[1] MPIRPC_Return(caller.rank=0,unique=1,length=-1)... 
[1] rpc_bootping(2)... 
[1] MPIRPC_Block(bootping)... 
[1] MPIRPC_Block_blob(bootping)... 
[1] MPIRPC_Call_blob(target.rank=2,name=bootping,args=(null),rpc->unique=1)... 
[0] MPIRPC_Retrieve(sender.rank=1,unique=1)... 
[0] Unblocked: bootping
[0] MPIRPC_Free(rpc->name=bootping,rpc->unique=1)... 
[0] listen_loop()... 
[2] MPIRPC_Request(name=bootping,caller.rank=1)... 
[2] MPIRPC_Return(caller.rank=1,unique=1,length=-1)... 
[2] rpc_bootping(3)... 
[1] MPIRPC_Retrieve(sender.rank=2,unique=1)... 
[1] Unblocked: bootping
[1] MPIRPC_Free(rpc->name=bootping,rpc->unique=1)... 
[2] MPIRPC_Block(bootping)... 
[2] MPIRPC_Block_blob(bootping)... 
[2] MPIRPC_Call_blob(target.rank=3,name=bootping,args=(null),rpc->unique=1)... 
[3] MPIRPC_Request(name=bootping,caller.rank=2)... 
[3] MPIRPC_Return(caller.rank=2,unique=1,length=-1)... 
[3] rpc_bootping(4)... 
[3] MPIRPC_Block(bootping)... 
[3] MPIRPC_Block_blob(bootping)... 
[3] MPIRPC_Call_blob(target.rank=4,name=bootping,args=(null),rpc->unique=1)... 
[4] MPIRPC_Request(name=bootping,caller.rank=3)... 
[4] MPIRPC_Return(caller.rank=3,unique=1,length=-1)... 
[4] rpc_bootping(-1)... 
[3] MPIRPC_Retrieve(sender.rank=4,unique=1)... 
[3] Unblocked: bootping
[3] MPIRPC_Free(rpc->name=bootping,rpc->unique=1)... 
[2] MPIRPC_Retrieve(sender.rank=3,unique=1)... 
[2] Unblocked: bootping
[2] MPIRPC_Free(rpc->name=bootping,rpc->unique=1)... 
running...
[5] cmpi_shutdown()... 
[5] cmpi_shutdown_list()... 
[5] cmpi_mode_first_client(nodes=5)... 
[5] DENSE_Shutdown()... 
[5] MPIRPC_Block(quit)... 
[5] MPIRPC_Block_blob(quit)... 
[5] MPIRPC_Call_blob(target.rank=0,name=quit,args=(null),rpc->unique=1)... 
[0] MPIRPC_Request(name=quit,caller.rank=5)... 
[0] handle_quit()... 
[0] MPIRPC_Return(caller.rank=5,unique=1,length=-1)... 
[5] MPIRPC_Retrieve(sender.rank=0,unique=1)... 
[5] Unblocked: quit
[5] MPIRPC_Free(rpc->name=quit,rpc->unique=1)... 
[5] DENSE_Shutdown()... 
[5] MPIRPC_Block(quit)... 
[5] MPIRPC_Block_blob(quit)... 
[5] MPIRPC_Call_blob(target.rank=1,name=quit,args=(null),rpc->unique=2)... 
[1] MPIRPC_Request(name=quit,caller.rank=5)... 
[1] handle_quit()... 
[1] MPIRPC_Return(caller.rank=5,unique=2,length=-1)... 
[5] MPIRPC_Retrieve(sender.rank=1,unique=2)... 
[5] Unblocked: quit
[5] MPIRPC_Free(rpc->name=quit,rpc->unique=2)... 
[5] DENSE_Shutdown()... 
[5] MPIRPC_Block(quit)... 
[5] MPIRPC_Block_blob(quit)... 
[5] MPIRPC_Call_blob(target.rank=2,name=quit,args=(null),rpc->unique=3)... 
[2] MPIRPC_Request(name=quit,caller.rank=5)... 
[2] handle_quit()... 
[2] MPIRPC_Return(caller.rank=5,unique=3,length=-1)... 
[5] MPIRPC_Retrieve(sender.rank=2,unique=3)... 
[5] Unblocked: quit
[5] MPIRPC_Free(rpc->name=quit,rpc->unique=3)... 
[5] DENSE_Shutdown()... 
[5] MPIRPC_Block(quit)... 
[5] MPIRPC_Block_blob(quit)... 
[5] MPIRPC_Call_blob(target.rank=3,name=quit,args=(null),rpc->unique=4)... 
[3] MPIRPC_Request(name=quit,caller.rank=5)... 
[3] handle_quit()... 
[3] MPIRPC_Return(caller.rank=5,unique=4,length=-1)... 
[5] MPIRPC_Retrieve(sender.rank=3,unique=4)... 
[5] Unblocked: quit
[5] MPIRPC_Free(rpc->name=quit,rpc->unique=4)... 
[5] DENSE_Shutdown()... 
[5] MPIRPC_Block(quit)... 
[5] MPIRPC_Block_blob(quit)... 
[5] MPIRPC_Call_blob(target.rank=4,name=quit,args=(null),rpc->unique=5)... 
[4] MPIRPC_Request(name=quit,caller.rank=5)... 
[4] handle_quit()... 
[4] MPIRPC_Return(caller.rank=5,unique=5,length=-1)... 
[5] MPIRPC_Retrieve(sender.rank=4,unique=5)... 
[5] Unblocked: quit
[5] MPIRPC_Free(rpc->name=quit,rpc->unique=5)... 
[5] cmpi_shutdown() done. 
[1] MPIRPC_Finalize()... 
[1] lru_table_destroy()... 
[3] MPIRPC_Finalize()... 
[3] lru_table_destroy()... 
[4] MPIRPC_Finalize()... 
[4] lru_table_destroy()... 
[5] MPIRPC_Finalize()... 
[5] Normal exit. 
[1] Normal exit. 
[3] Normal exit. 
[4] Normal exit. 
[0] MPIRPC_Finalize()... 
[2] MPIRPC_Finalize()... 
[2] lru_table_destroy()... 
[0] lru_table_destroy()... 
[2] Normal exit. 
[0] Normal exit. 
